dotenv.go

```diff
@@ -0,0 +1,218 @@
+package dotenv
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+	"unicode"
+)
+
+const (
+	posKey int = iota
+	posVal
+)
+
+// ErrInvalidln indicates an invalid line
+var ErrInvalidln = errors.New("invalid line")
+
+// ErrEmptyln indicates an empty line
+var ErrEmptyln = errors.New("empty line")
+
+// ErrCommentln indicates a comment line
+var ErrCommentln = errors.New("comment line")
+
+var varRE = regexp.MustCompile(`\${\w+}`)
+
+// ReadFile reads an env file at a given path, and return values as a map.
+func ReadFile(path string) (map[string]string, error) {
```

Comment:
Does `ReadFile` _need_ to be exported?  Is its equivalent function in the Ruby lib part of the API?

Reply:
Ah, good point. Since `Load` changed to what it is now, I don't believe this does need to be exported.